packages/teams-js/src/internal/communication.ts

```diff
@@ -305,6 +312,70 @@ function processMessage(evt: DOMMessageEvent): void {
 
 const shouldProcessMessageLogger = communicationLogger.extend('shouldProcessMessage');
 
+function sendValidateDomainRequest(messageOrigin: URL): Promise<boolean> {
+  const messageOriginUrl = sanitizeAndStrigifyURL(messageOrigin);
+  return sendAndUnwrap('validateDomains', messageOriginUrl);
+}
+
+export function validateOrigin(messageOrigin: URL): boolean {
```

Comment:
>boolean [](http://example.com/codeflow?start=52&length=7)

This should return a `Promise<boolean>` since one of the function calls in here (`sendValidateDomainRequest`) is returning a `Promise` -- this ensures that any callers know there is async stuff that needs to be waited on and can wait on it.